src/leer.java

```diff
@@ -8,264 +8,210 @@
+	            }
+	    }
+	    catch (IOException e) {
+	          e.printStackTrace();
+	    }
+	    finally {
+	        result= calculate.prepare_result(i,id_ride_final,fare);
+		    if (buffer_read != null) {
+	           try {
+	               buffer_read.close();
+	           }
+	           catch (IOException e) {
+	                e.printStackTrace();
+			   }
+	        }
+		    try {
+                String route = "/Users/a.manning/Downloads/result.txt";
+                String content = result;
+			    File file = new File(route);
+                // if the file doesn't exist is created
+                if (!file.exists()) {
+                   file.createNewFile();
+                }
+			    FileWriter fw = new FileWriter(file);
+                BufferedWriter bw = new BufferedWriter(fw);
+		        bw.write(content);
+                bw.close();
+            }
+		    catch (Exception e) {
+                  e.printStackTrace();
+			}
+        }
+	}
+public  double calculate_speed (String lat,
+						  String lng,
+						  String lat_line2,
+						  String lng_line2,
+						  String id_ride,
+						  String timestamp,
+						  String timestamp_line2,
+						  int i)
+{
+	double a,c,distance,U,delta_latitude,delta_longitude,delta_time;
 	int R =6371;
+
 	leer calculate = new leer();
 	
-    float id_ridea = Float.parseFloat(id_ride);
-    float lata = Float.parseFloat(lat);
-    float latb = Float.parseFloat(lat1);
-    float lnga = Float.parseFloat(lng);
-    float lngb = Float.parseFloat(lng1);
-    Integer timestampa = Integer.parseInt(timestamp);
-    Integer timestampb = Integer.parseInt(timestamp1);
-	
-//calculating distance
-    deltlat=Math.toRadians(latb - lata) ;
-    deltlng= Math.toRadians(lngb - lnga);
-    delttime= (timestampb - timestampa);
-    a = Math.pow(Math.sin(deltlat/2),2)+ Math.cos(lata)*Math.cos(latb)* Math.pow(Math.sin(deltlng/2),2);
-    c = 2* Math.atan2(((Math.sqrt(a))),(Math.sqrt(1-a)));
+    float new_id_ride = Float.parseFloat(id_ride);
+    float new_lat = Float.parseFloat(lat);
+    float new_lat_line2 = Float.parseFloat(lat_line2);
+    float new_lng = Float.parseFloat(lng);
+    float new_lng_line2 = Float.parseFloat(lng_line2);
+    int new_timestamp = Integer.parseInt(timestamp);
+    int new_timestamp_line2 = Integer.parseInt(timestamp_line2);
+
+    delta_latitude = Math.toRadians(new_lat_line2 - new_lat) ;
+    delta_longitude = Math.toRadians(new_lng_line2 - new_lng);
+    delta_time = (new_timestamp_line2 - new_timestamp);
+    a = Math.pow(Math.sin(delta_latitude/2),2) +
+			Math.cos(new_lat) * Math.cos(new_lat_line2) *
+					Math.pow(Math.sin(delta_longitude/2),2);
+	c = 2 * Math.atan2(((Math.sqrt(a))),(Math.sqrt(1-a)));
     distance = (R*c);
-    U = (distance/delttime)*3600;
-    
-	calculate.fare_rules(distance, U, deltlat, deltlng, delttime, id_ridea, timestampa, i);
+	U = (distance/delta_time)*3600;
+	calculate.fare_rules(distance, U, delta_time, new_id_ride, new_timestamp, i);
```

Comment:
This is an excelent example of idiomatic naming. If you see, you can read as Calculate Fare Rule

Reply:
great!! i am trying to apply what i've learned from the videos, that is why i tried to add names that reflect better what is the purpose of the method :D